daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -889,4 +889,106 @@ object TransactionCoder {
       Right(None)
     }
 
+  private[transaction] def encodeVersioned(
+      version: TransactionVersion,
+      unversioned: ByteString,
+  ): ByteString = {
+    val byteStringOutput = ByteString.newOutput
+    val out = CodedOutputStream.newInstance(byteStringOutput)
+    out.writeInt32NoTag(version.index)
+    out.writeBytesNoTag(unversioned)
+    out.flush()
+    byteStringOutput.toByteString
+  }
```

Comment:
I did not use protobuf to wrap the message with its version.
This let us more foredoom to change the serialization  (e.g. move complelty away from proto for efficiency or security reason)

Reply:
I would let you be the judge what is preferable here. 

However I need to point out that technically speaking the inverse of the given justification is true: if you start with proto encoding, you can evolve i away from it in subsequent version. If you start with manual encoding though, you can never put the genie back into the bottle.
 
It is because it is always possible to decompose protobuf by hand, but protobuf will never be able to parse the string as implemented here

Reply:
Why not use the encoding matching
```
message VersionedPayload {
  int32 version = 1;
  bytes payload = 2
}
```
which uses two bytes more than your encoding, but leaves the freedom of decoding this message using protobuf; i.e., just add the two tags to the two invocations you have.

Reply:
c37fc4c4da6ca934dd8ebe61b236c00c310d34ce